libs/gltfio/src/ResourceLoader.cpp

```diff
@@ -669,22 +674,30 @@ void ResourceLoader::Impl::cancelTextureDecoding() {
 }
 
 void ResourceLoader::Impl::createTextures(FFilamentAsset* asset, bool async) {
-    // Create new texture objects if they are not cached and kick off decoding jobs.
     mRemainingTextureDownloads = 0;
-    size_t textureIndex = 0;
-    for (FFilamentAsset::TextureInfo& info : asset->mTextures) {
-        // There might be no bindings yet (e.g. if the user is preloading a zero-instance asset)
-        // in which case we still need to trigger texture creation.
-        getOrCreateTexture(asset, textureIndex, info.flags);
-
-        // For each binding to a material instance, check if the texture exists yet; if so,
-        // call setParameter(...) on the material.
-        for (const TextureSlot& slot : info.bindings) {
-            if (Texture* texture = getOrCreateTexture(asset, textureIndex, info.flags)) {
-                asset->applyTextureBinding(textureIndex, slot);
+
+    // Create new texture objects if they are not cached and kick off decoding jobs.
+    for (size_t textureIndex = 0, n = asset->mTextures.size(); textureIndex < n; ++textureIndex) {
+        FFilamentAsset::TextureInfo& info = asset->mTextures[textureIndex];
+        auto [texture, cacheResult] = getOrCreateTexture(asset, textureIndex, info.flags);
+        if (texture == nullptr) {
+            if (cacheResult == CacheResult::NOT_READY) {
+                mRemainingTextureDownloads++;
```

Comment:
that's single threaded correct?

Reply:
Yes all reads and writes to `mRemainingTextureDownloads` occur on the same thread.